test/03_USM.test.js

```diff
@@ -29,127 +43,280 @@ contract('USM', (accounts) => {
-        await expectRevert(usm.mint(user1, user2, oneEth, { from: user1 }), 'Fund before minting')
+        await expectRevert(usm.mint(user1, user2, oneEth, { from: user1 }), "Fund before minting")
       })
 
-      it('allows minting FUM', async () => {
+      it("allows minting FUM", async () => {
         const fumBuyPrice = (await usm.fumPrice(sides.BUY))
         const fumSellPrice = (await usm.fumPrice(sides.SELL))
+        fumBuyPrice.toString().should.equal(fumSellPrice.toString()) // We should start with the buy & sell FUM prices equal
 
         await usm.fund(user1, user2, oneEth, { from: user1 })
-        const fumBalance = (await fum.balanceOf(user2))
-        fumBalance.toString().should.equal(oneEth.mul(priceWAD).div(WAD).toString()) // after funding we should have eth_price fum per eth passed in
+        const ethPool2 = (await weth.balanceOf(usm.address))
+        ethPool2.toString().should.equal(oneEth.toString())
+
+        const fumBalance2 = (await fum.balanceOf(user2))
+        // The very first call to fund() is special: the initial price of $1 is used for the full trade, no sliding prices.  So
+        // 1 ETH @ $250 should buy 250 FUM:
+        fumBalance2.toString().should.equal(wadMul(oneEth, priceWAD).toString())
+
+        const fundDefundAdj2 = (await usm.fundDefundAdjustment())
+        fundDefundAdj2.toString().should.equal(WAD.toString()) // First call to fund() doesn't move the fundDefundAdjustment
 
-        const newEthPool = (await weth.balanceOf(usm.address))
-        newEthPool.toString().should.equal(oneEth.toString())
+        const fumBuyPrice2 = (await usm.fumPrice(sides.BUY))
+        fumBuyPrice2.toString().should.equal(fumBuyPrice.toString()) // First fund() (w/o sliding prices) doesn't change FUM price
+        const fumSellPrice2 = (await usm.fumPrice(sides.SELL))
+        fumSellPrice2.toString().should.equal(fumSellPrice.toString())
 
-        const newFumBuyPrice = (await usm.fumPrice(sides.BUY))
-        newFumBuyPrice.toString().should.equal(fumBuyPrice.toString()) // Funding doesn't change the fum price
-        const newFumSellPrice = (await usm.fumPrice(sides.SELL))
-        newFumSellPrice.toString().should.equal(fumSellPrice.toString())
+        await usm.fund(user1, user2, oneEth, { from: user1 })
+        const ethPool3 = (await weth.balanceOf(usm.address))
+        ethPool3.toString().should.equal(oneEth.mul(TWO).toString())
+
+        const fumBalance3 = (await fum.balanceOf(user2))
+        // Calls to fund() after the first start using sliding prices.  In particular, following the math in USM.fumFromFund(),
+        // with the second call here 2xing the pool (1 ETH -> 2) and the initial FUM price of $1 = 0.004 ETH, the FUM added should
+        // be 1 (current ETH pool) / 0.004 * (1 - 1 / 2) = 125, ie, the user's FUM balance should increase by a factor of 3/2:
+        fumBalance3.toString().should.equal(wadMul(oneEth, priceWAD).mul(THREE).div(TWO).toString())
```

Comment:
As before, the expected FUM increase should be calculated in the test, instead of hardcoded to `3/2`.

Also, please store the FUM supply in a variable, instead of using `wadMul(oneEth, priceWAD)`

Reply:
Hmmm, I guess this part I haven't yet parameterized.  What I've done so far is this:
1. user1 and user2 both start with certain ("parameterized") amounts of ETH.
2. user1 spends all their ETH `fund()`ing (with the FUM going to user2).
3. user2 spends all their ETH `mint()`ing (with the USM going to user1).

But this doesn't parameterize everything: eg, the `3/2` here follows because this particular test calls `fund()` twice (since the first `fund()` is a special case that doesn't use sliding prices, which we do want to test!), and I originally just made the test call both times with the same input amount of ETH.  If the second `fund()` call passed in a different amount of ETH from the first, this `3/2` would change.  So I guess it would be a matter of adding a third parameter - not just "how much ETH user1 passes to `fund()`" and "how much ETH user2 passes to `mint()`", but also "how much ETH user1 passes to their _second_ call to `fund()`"?

If that makes sense to you, I'll add that third parameter.  The math actually isn't bad.